Node.js - fuzzy search in next release (#1459)
cap-js/docs: guides/providing-services.md

Fuzzy search is a fault-tolerant search feature of SAP HANA Cloud, which returns records even if the search term contains additional characters, is missing characters, or has typographical errors.

> If you run in [`HEX` optimization mode](../java/cqn-services/persistence-services#sql-optimization-mode) on SAP HANA Cloud, you can enable fuzzy search in the *application.yaml* and configure the default fuzziness in the range [0.0, 1.0]. The value 1.0 enforces exact search. The default fuzziness is 0.8.

Reviewer: How is that configured in Node.js? We could as well profile the sentence and add what's relevant for Node.js here.

Author: We do not have an hex optimization mode. hex engine is always used.

Reviewer: ```suggestion
If you run in [`HEX` optimization mode (default for Node.js)](../java/cqn-services/persistence-services#sql-optimization-mode) on SAP HANA Cloud, you can enable fuzzy search and configure the default fuzziness in the range [0.0, 1.0]. The value 1.0 enforces exact search. The default fuzziness is 0.8.

- Java: <Config java>cds.sql.hana.search.fuzzy = true</Config>
- Java: <Config java>cds.sql.hana.search.fuzzinessThreshold = 0.9</Config>
- Node.js:<Config>cds.hana.fuzzy = 0.9</Config>
```

Reviewer: The default fuzziness is different in Node.js, isn't it? It's 0.7 and not 0.8, right?

> <span markdown="1" class="impl java">
> 
> ```yml
> cds.sql.hana.search
>    fuzzy: true
>    fuzzinessThreshold: 0.9
> ```
> 
> </span>
> 
> <span markdown="1" class="impl node">
> 
> ```json
> "cds": {
>   "hana": {
>     "fuzzy": 0.9
>   }
> }
> ```
> 
> </span>
> 

Reviewer: ```suggestion

```

> Override the fuzziness for elements, using the `@Search.fuzzinessThreshold` annotation:

Reviewer: Is this annotation any different for Node.js?

Supported wildcards are '*' matching zero or more characters and '?' matching a single character. You can escape wildcards using '\\'.
:::

::: warning

> Fuzzy search on SAP HANA Cloud requires the [`HEX` optimization mode](../java/cqn-services/persistence-services#sql-optimization-mode).

Reviewer: ```suggestion
Fuzzy search on SAP HANA Cloud requires the [`HEX` optimization mode (default for Node.js)](../java/cqn-services/persistence-services#sql-optimization-mode).
```
